### PR TITLE
core: Refactor code quality issues

### PR DESCRIPTION
**Pull request by akshgpt7**
## Description

Hey, I'm a Developer Outreach at [DeepSource](https://deepsource.io/) and ran DeepSource analysis on my fork of the repo. It found some interesting [code quality and performance improvements](https://deepsource.io/gh/akshgpt7/ceph-ansible/) to consider.

This PR fixes a few of the issues detected for you to assess if it is right for you.
Happy to provide the tweaks separately otherwise :)

## Summary of changes
- Use literals instead of function calls to create data structure.
- Refactor unnecessary list comprehension.
- Simplify `if` statement.
- Refactor useless `else` block in the loop.
- Add `.deepsource.toml` file for continuous analysis on bug risks/performance/code-quality issues on new changes.

### CODE REVIEW

**Comment by guits** [comment]
Hi @akshgpt7 

Interesting patch, thanks for this contribution.

Do you mind adding a commit message and a sign-off to your commit please?

Thanks!

**Comment by akshgpt7** [comment]
Thanks @guits, I've added a commit message, and signed it off.

**Comment by guits** [comment]
jenkins test centos-non_container-all_daemons

**Comment by guits** [comment]
jenkins test centos-non_container-collocation

**Comment by guits** [comment]
jenkins test centos-non_container-all_in_one

**Comment by guits** [comment]
jenkins test centos-non_container-purge

**Comment by guits** [comment]
jenkins test centos-non_container-switch_to_containers

**Comment by guits** [comment]
@akshgpt7 FYI, i've fixed the few conflicts

File: plugins/actions/config_template.py [hunk 1]
```diff
@@ -179,8 +179,8 @@ def _do_write(section_name, section, section_bool=False):
                 _write_comments(section_name, optname=key)
                 self._write_check(fp, key=key, value=value,
                                   section=section_bool)
-            else:
-                fp.write("\n")
+
```
Comment (guits) [inline plugins/actions/config_template.py:182]:
```suggestion
```

File: plugins/actions/config_template.py [hunk 2]
```diff
@@ -453,8 +453,8 @@ def return_config_overrides_ini(self,
                             'data. Sections are case sensitive.'
                         )
                         raise errors.AnsibleModuleError(error_msg)
-        else:
-            config_object.close()
+
```
Comment (guits) [inline plugins/actions/config_template.py:456]:
```suggestion
```

**Review by guits — CHANGES_REQUESTED** [review]

**Review by guits — APPROVED** [review]

**Comment by guits** [comment]
jenkins test centos-non_container-switch_to_containers

**Comment by guits** [comment]
jenkins test centos-container-all_in_one

**Comment by guits** [comment]
jenkins test centos-non_container-switch_to_containers

**Comment by akshgpt7** [comment]
@guits Is there anything else that needs to be done from my side?

**Comment by guits** [comment]
jenkins test centos-non_container-switch_to_containers

**Comment by guits** [comment]
> @guits Is there anything else that needs to be done from my side?

no, so failure we are seeing are not related to your patch, I'm working on fixing the ci

**Comment by akshgpt7** [comment]
@guits Thanks for merging this. I'd just like to inform you that the issues fixed here were detected by running DeepSource analysis on the repo. If you like, you can activate analysis for your repository to detect such code quality issues/bug risks on the fly for every change made. You can also use the Autofix feature to fix them with one click.
The `.deepsource.toml` file you merged will only take effect if you activate analysis for this repo.

Here's what you can do if you wish to activate DeepSource to continuously analyze your repository:
- [Sign up](https://deepsource.io/signup) on DeepSource and activate analysis for this repository.
- Create `.deepsource.toml` configuration which you can use to configure your analysis settings (this PR already added that, but feel free to edit it anytime).
- Track/Check analysis [here](https://deepsource.io/gh/ceph/ceph-ansible/).

If you have any doubts or questions, you can check out the [docs](https://deepsource.io/docs/), or feel free to reach out :)

**Comment by akshgpt7** [comment]
@guits do you need any help with activating DeepSource?
I ask so because the `.deepsource.toml` file that you merged will only take effect if analysis is activated.

**Pull request merged**